File: lib/std/java/java/nio/file/OutputFile.java
package java.nio.file;

public class OutputFile {

    private String fileName;

    OutputFile(String filename) {
        this.fileName = filename;
        open(filename);
    }

    public native void open(String filename);
    public native void close();
    public native void writeLine(String text);
    public native void write(String text);
}

File: lib/mrjvm/native/file/output_file.rb
class OutputFile
  attr_reader :filename

  def initialize(filename)
    @filename = filename
    @file = File.open(@filename, 'w')
  end

  def write_line(line)
    @file.write(line + "\n")
  end

  def write(text)
    @file.write(text)
  end

  def close
    @file.close
  end
end
